plugins/org.jboss.tools.openshift.io.ui/src/main/java/org/jboss/tools/openshift/io/internal/ui/DefaultLoginProvider.java

```diff
@@ -25,23 +31,80 @@
  */
 public class DefaultLoginProvider implements LoginProvider {
 
+	private class LoginJob extends UIJob {
+		private boolean runninginUI = false;
+		private boolean shouldRun = true;
+		private ICluster cluster;
+		private IAccount account;
+		private LoginResponse response;
+		
+		public LoginJob(ICluster cluster, IAccount account) {
+			super("Logging to OpenShift.io");
+			this.cluster = cluster;
+			this.account = account;
+		}
+
+		@Override
+		public IStatus runInUIThread(IProgressMonitor monitor) {
+			if (shouldRun && !monitor.isCanceled()) {
```

Comment:
It is there for safety because cancelling the job may be delayed due to lock mechanisms and as this job will show some UI I though it was important to be safe